marshal_test.go

```diff
@@ -1,75 +1,92 @@
-package sshkeys
+package sshkeys_test
 
 import (
 	"crypto/rand"
 	"testing"
 
-	"golang.org/x/crypto/ssh"
-
+	"github.com/ScaleFT/sshkeys"
 	"github.com/ScaleFT/sshkeys/testdata"
 	"github.com/stretchr/testify/require"
+	"golang.org/x/crypto/ssh"
 )
 
 func testSigners(t *testing.T, name string, a ssh.Signer, b ssh.Signer) {
 	require.Equal(t, a.PublicKey().Marshal(), b.PublicKey().Marshal())
 
 	sign := []byte("hello world")
 	sig, err := a.Sign(rand.Reader, sign)
-	require.NoError(t, err, "signer failed for "+name)
+	require.NoError(t, err, "signer failed for %s", name)
 
 	err = b.PublicKey().Verify(sign, sig)
-	require.NoError(t, err, "verify failed for "+name)
+	require.NoError(t, err, "verify failed for %s", name)
 }
 
-func TestMarshal(t *testing.T) {
+func TestMarshalOldFormat(t *testing.T) {
 	password := []byte("gopher")
 	for _, k := range testdata.PEMEncryptedKeys {
-		pk, err := ParseEncryptedRawPrivateKey(k.PEMBytes, []byte(k.EncryptionKey))
-		require.NoError(t, err, "error parsing "+k.Name)
-		require.NotNil(t, pk, "nil return from parsing "+k.Name)
+		// ed25519 is only specified in the new format
+		if k.Name == "ed25519-openssh-encrypted-aes256-cbc" || k.Name == "ed25519-openssh-encrypted-aes256-ctr" {
```

Comment:
maybe we should just have two test arrays?  Or make a format_type, rather than looking for specific names?

Reply:
I keep the same approach that we have before, but totally agree on split it in two arrays. Do you suggest to refactor it in the same PR o move in a different one?